### PR TITLE
WIP: Add integration tests for Lightning support

### PR DESCRIPTION
**Pull request by dnanuti**
## Description

This PR adds E2E tests for Lightning checkpoints. CI tasks and imports need to be double checked.
## Related items

- https://github.com/awslabs/s3-connector-for-pytorch/pull/132

## Testing

- Locally ran against personal S3 bucket
--------
By submitting this pull request, I confirm that my contribution is made under the terms of BSD 3-Clause License and I agree to the terms of the [LICENSE](https://github.com/awslabs/s3-connector-for-pytorch/blob/main/LICENSE).

### CODE REVIEW

File: s3torchconnector/tst/e2e/models/lightning_transformer.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
+#  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#  // SPDX-License-Identifier: BSD
+from typing import Dict, Any
+
+import lightning as L
```
Comment (muddyfish) [inline s3torchconnector/tst/e2e/models/lightning_transformer.py:5]: Why not do a from import LightningModule here given that we're only using this once?

File: s3torchconnector/tst/e2e/models/net.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
+import torch
```
Comment (muddyfish) [inline s3torchconnector/tst/e2e/models/net.py:1]: Missing copyright. Though are these files something Amazon owns, or did you copy/paste this from an example?

File: s3torchconnector/tst/e2e/test_e2e_s3_lightning_checkpoint.py [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+    dataloader = DataLoader(dataset)
+    model = LightningTransformer(vocab_size=dataset.vocab_size)
+    trainer = L.Trainer(fast_dev_run=3)
+    trainer.fit(model=model, train_dataloaders=dataloader)
+    checkpoint_name = "lightning_module_training_checkpoint.pt"
+    s3_uri = f"{checkpoint_directory.s3_uri}/{checkpoint_name}"
+    trainer.save_checkpoint(s3_uri)
+    lightning_checkpoint = S3LightningCheckpoint(region=checkpoint_directory.region)
+    loaded_model = lightning_checkpoint.load_checkpoint(s3_uri)
+    for model_item, loaded_item in zip(
```
Comment (muddyfish) [inline s3torchconnector/tst/e2e/test_e2e_s3_lightning_checkpoint.py:61]: Do you think `(model_key, model_value), (loaded_key, loaded_item) in ...` would look better, and then you'd not have to do the index lookups later

File: s3torchconnector/tst/e2e/models/net.py (hunk 2 above)
Comment (dnanuti) [reply s3torchconnector/tst/e2e/models/net.py:1]: This was previously added as part of the checkpoint work, but I think it's from an example too - https://pytorch.org/tutorials/beginner/blitz/neural_networks_tutorial.html, that's why I removed the copyright, but missed removing it in my class.

**Review by IsaevIlya — COMMENTED** [review]
Could we cover a few more cases, please:
- case for using `S3LightningCheckpoint` wrapped in `AsyncCheckpointIO`
- one more case for usage as plugin with callback to test `save` and `delete` path at the same time:
```
s3_checkpoint_io = S3LightningCheckpoint("us-east-1")
checkpoint_callback = ModelCheckpoint(
    dirpath="s3://bucket_name/key_prefix/", 
    save_top_k=1,
    every_n_epochs=1,
    monitor="val_loss",
    filename="checkpoint-{epoch:02d}-{val_loss:.2f}.ckpt"
)
trainer = Trainer(
    plugins=[s3_checkpoint_io],
    callbacks=[checkpoint_callback]
)
trainer.fit(model)
# after finishing there should be only one checkpoint left
# name of the checkpoint should be the same as checkpoint_callback.best_model_path
```
- case for loading data from checkpoint via plugin:
```
trainer = pl.Trainer(
    plugins=[s3_checkpoint_io], 
    resume_from_checkpoint='s3://bucket_name/key_prefix/checkpoint.ckpt'
)
```

**Comment by dnanuti** [comment]
This was automatically closed due to a bad rebase.

**Pull request merged**